mne/report/report.py

```diff
@@ -1069,92 +688,812 @@ def add_custom_js(self, js):
+                   tags=('custom-figure',), replace=False):
+        """Add figures to the report.
+
+        Parameters
+        ----------
+        fig : matplotlib.figure.Figure | mlab.Figure | array |
+              collection of matplotlib.figure.Figure |
+              collection of mlab.Figure | collection of array
+
+            One or more figures to add to the report. All figures must be an
+            instance of :class:`matplotlib.figure.Figure`,
+            :class:`mayavi.core.api.Scene`, or :class:`numpy.ndarray`. If
+            multiple figures are passed, they will be added as "slides"
+            that can be navigated using buttons and a slider element.
+        title : str
+            The title corresponding to the figure(s).
+        caption : str | collection of str | None
+            The caption(s) to add to the figure(s).
+        %(report_image_format)s
+        %(report_tags)s
+        %(report_replace)s
+
+        Notes
+        -----
+        .. versionadded:: 0.24.0
+        """
+        tags = tuple(tags)
+        if image_format is None:
+            image_format = self.image_format
+
+        if hasattr(fig, '__len__') and not isinstance(fig, np.ndarray):
+            figs = tuple(fig)
         else:
-            # Append new record
-            self.fnames.append(fname)
-            self._sectionlabels.append(sectionlabel)
-            self.html.append(html)
+            figs = (fig,)
 
+        for fig in figs:
+            if _check_path_like(fig):
+                raise TypeError(
+                    f'It seems you passed a path to `add_figure`. However, '
+                    f'only Matplotlib figures, Mayavi scenes, and NumPy '
+                    f'arrays are accepted. You may want to try `add_image` '
+                    f'instead. The provided path was: {fig}'
+                )
+        del fig
+
+        if isinstance(caption, str):
+            captions = (caption,)
+        elif caption is None:
+            captions = [f'Figure {i+1}' for i in range(len(figs))]
+        else:
+            captions = tuple(caption)
+
+        del caption
+
+        assert figs
+        if len(figs) == 1:
+            img = _fig_to_img(fig=figs[0], image_format=image_format)
+            dom_id = self._get_dom_id()
+            html = _html_image_element(
+                img=img, div_klass='custom-image', img_klass='custom-image',
+                title=title, caption=captions[0], show=True,
+                image_format=image_format, id=dom_id, tags=tags
+            )
+        else:
+            html, dom_id = self._render_slider(
+                figs=figs, title=title, captions=captions, start_idx=0,
+                image_format=image_format, tags=tags
+            )
+
+        self._add_or_replace(
+            dom_id=dom_id,
+            name=title,
+            tags=tags,
+            html=html,
+            replace=replace
+        )
+
+    @fill_doc
```

Comment:
This should be deprecated no? Please make sure that doc strings of all mention the deprecation so it appears in the API page what method is deprecated.

Reply:
In fact, I [removed](fcec676) the `@deprecated` decorator and started "manually" emitting `DeprecationWarning`s because the decorator was causing issues with numpydoc. I posted this on Discord:

The following signature, for example:

```
    @deprecated(extra='Use `Report.add_image` instead')
    def add_images_to_section(self, fnames, captions, scale=None,
                              section='custom', comments=None, replace=False):
```

Would cause the following warning during the doc build:

```
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne.Report.add_images_to_section':
  SS03: Summary does not end with a period
  PR01: Parameters {'**kwargs', '*args'} not documented
  PR02: Unknown parameters {'replace', 'section', 'captions', 'fnames', 'scale', 'comments'}
  RT01: No Returns section found
```

Any suggestions on how to fix this would be appreciated!

Reply:
@agramfort I've now hacked my way around this in 6d8f9ac

Not pretty but it works.